### PR TITLE
Review fixes

### PR DESCRIPTION
**Pull request by thejacer87**

### CODE REVIEW

**Comment by thejacer87** [comment]
fixes issue brought up here:
* e0d92bd2b4
* #5

File: src/ClientBase.php [hunk 1]
```diff
@@ -108,14 +108,14 @@ public function post($endpoint, array $headers = [], $payload)
...
-            $message = 'A username and a password are required for authenticated to the Canada Post API.';
+        if (!isset($config['username']) || !isset($config['password']) || !isset($config['customer_number'])) {
+            $message = 'A username, password and customer number are required for authenticated to the Canada Post API.';
             throw new \InvalidArgumentException($message);
         }
 
         $this->username = $config['username'];
         $this->password = $config['password'];
-        $this->customerNumber = $config['customerNumber'];
+        $this->customerNumber = $config['customer_number'];
```
Comment (krystalcode) [inline src/ClientBase.php]: The customer number is actually not a requirement and there are several endpoints that can be called without it. However, let's leave it like this right now so we can make a release and we can improve on this later. Each endpoint should be validating its required arguments.

File: src/Rating.php [hunk 2]
```diff
@@ -8,37 +8,35 @@ class Rating extends ClientBase
...
-      return $response;
+        $response = $this->post(
+          "rs/ship/price",
+          [
+            'Content-Type' => 'application/vnd.cpc.ship.rate-v3+xml',
+            'Accept' => 'application/vnd.cpc.ship.rate-v3+xml',
+          ],
+          $payload
+        );
+        return $response;
```
Comment (krystalcode) [inline src/Rating.php]: I assume you didn't just copy/paste the review code without testing this.

Comment (thejacer87) [reply src/Rating.php]: yup, tested with rates and it was good

**Pull request merged**